mantidproject/mantid
# Unreliable muon pyqt tests

**Description of work.**

Some Muon GUI tests currently appear to be failing sporadically on the build servers. Upon investigation this appears to be at least partially because some of the tests are throwing exceptions which are not caught by the testing framework. 

This is causing some to fail in QT5 but not QT4 due to this change in functionality https://stackoverflow.com/questions/43641638/unhandled-exceptions-in-pyqt5

This pull request updates the GuiTest class in order to force the tests to fail when an exception is thrown and adds some extra clean-up methods for safety.

**To test:**

 * Check that tests pass and code changes look sensible.

*There is no associated issue.*


*This does not require release notes* because this is a change which only effects the test suite


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
